templates/cpp-template-default/CMakeLists.txt

```diff
@@ -37,6 +37,11 @@ include(CocosBuildHelpers)
 set(BUILD_CPP_TESTS OFF CACHE BOOL "turn off build cpp-tests")
 set(BUILD_LUA_LIBS OFF CACHE BOOL "turn off build lua related targets")
 set(BUILD_JS_LIBS OFF CACHE BOOL "turn off build js related targets")
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(DEBUG_MODE ON CACHE BOOL "turn on debug mode")
+else()
+  set(DEBUG_MODE OFF CACHE BOOL "turn off debug mode")
+endif()
```

Comment:
I'm searching for occurrences of "DEBUG_MODE" in Cocos2dx v3.4 (That's the one I'm using currently.) and as you said cocos2d only uses "DEBUG_MODE" to control "CMAKE_BUILD_TYPE". Is that really necessary? CMake also has `RelWithDebInfo` and `MinSizeRel` options for `CMAKE_BUILD_TYPE`. Wouldn't it be better to just use `CMAKE_BUILD_TYPE`? Is there an advantage using `DEBUG_MODE` instead of `CMAKE_BUILD_TYPE`?